[Bug]: Module not found: Can't resolve 'window-or-global' in .. carbon-components-react\es\internal

### Package

carbon-components-react

### Browser

_No response_

### Package version

^7.34.0

### React version

^16.8.6

### Description

This happened after I updated an own dependency. Maybe was a patch applied in one of your packages in the dependency tree? Maybe one of that removed `window-or-global` from the dependencies?

When I want to start my react app, it fails to compile:

```
Failed to compile.

./node_modules/carbon-components-react/es/internal/FloatingMenu.js
Module not found: Can't resolve 'window-or-global' in 'C:\Users\...\node_modules\carbon-components-react\e
s\internal'
Error from chokidar (C:\node_modules): Error: EBUSY: resource busy or locked, lstat 'C:\DumpStack.log.tmp'
Error from chokidar (C:\node_modules): Error: EBUSY: resource busy or locked, lstat 'C:\DumpStack.log.tmp'
```
I also tried to remove node_modules and yarn.lock, and reinstalled all packages.

Due to the chokidar errors I tried to clean the yarn cache and rebooted the pc. Problem persists.

**EDIT**
In yarn.lock I see several minor version updates were applied to carbon packages. Also I see that the `window-or-global@^1.0.1:` block was completely removed without a replacement of its version.

### CodeSandbox example

-

### Steps to reproduce

-

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

--- a/packages/react/package.json
+++ b/packages/react/package.json
@@ -63,7 +63,8 @@
     "prop-types": "^15.7.2",
     "react-is": "^17.0.2",
     "use-resize-observer": "^6.0.0",
-    "wicg-inert": "^3.1.1"
+    "wicg-inert": "^3.1.1",
+    "window-or-global": "^1.0.1"
   },
   "devDependencies": {
     "@babel/cli": "^7.16.7",

--- a/packages/type/scss/modules/_styles.scss
+++ b/packages/type/scss/modules/_styles.scss
@@ -9,6 +9,7 @@
 
 @use 'sass:map';
 @use 'sass:math';
+@use '@carbon/grid/scss/modules/config' as gridconfig;
 @use '@carbon/grid/scss/modules/breakpoint' as grid;
 @use 'font-family';
 @use 'scale';
@@ -716,7 +717,7 @@ $tokens: (
 /// @param {Map} $breakpoints [$grid-breakpoints] - Custom breakpoints to use
 /// @access public
 /// @group @carbon/type
-@mixin fluid-type($type-styles, $breakpoints: grid.$grid-breakpoints) {
+@mixin fluid-type($type-styles, $breakpoints: gridconfig.$grid-breakpoints) {
   // Include the initial styles for the given token by default without any
   // media query guard. This includes `font-size` as a fallback in the case
   // that a browser does not support `calc()`
@@ -745,7 +746,7 @@ $tokens: (
 @mixin fluid-type-size(
   $type-styles,
   $name,
-  $breakpoints: grid.$grid-breakpoints
+  $breakpoints: gridconfig.$grid-breakpoints
 ) {
   // Get the information about the breakpoint we're currently working in. Useful
   // for getting initial width information
@@ -846,7 +847,11 @@ $custom-property-prefix: 'cds' !default;
 /// @param {Map} $breakpoints [$grid-breakpoints] - Provide a custom breakpoint map to use
 /// @access public
 /// @group @carbon/type
-@mixin type-style($name, $fluid: false, $breakpoints: grid.$grid-breakpoints) {
+@mixin type-style(
+  $name,
+  $fluid: false,
+  $breakpoints: gridconfig.$grid-breakpoints
+) {
   @if not map.has-key($tokens, $name) {
     @error 'Unable to find a token with the name: `#{$name}`';
   }

--- a/packages/type/scss/modules/_styles.scss
+++ b/packages/type/scss/modules/_styles.scss
@@ -9,7 +9,7 @@
 
 @use 'sass:map';
 @use 'sass:math';
-@use '@carbon/grid/scss/modules/config' as grid;
+@use '@carbon/grid/scss/modules/breakpoint' as grid;
 @use 'font-family';
 @use 'scale';
 

